js/pages/questionnaire.js

```diff
@@ -56,32 +82,40 @@ const importQuest = async () => {
 
 /**
  * Questionnaire directs the survey module handling throughout the survey loading process.
- * Errors from children are caught here. The loading animation is shown and hidden here.
+ * Errors from the module loading process are caught here. The loading animation is shown and hidden here.
  * @param {string} moduleId - The ID of the survey module the participant clicked to start.
  */
+
 export const questionnaire = async (moduleId) => {
     try {
         showAnimation();
 
         if (!moduleId) {
             throw new Error('No module ID on start survey click.');
         }
-        
-        const [responseData] = await Promise.all([
+
+        const [participantResponse, appSettingsResponse] = await Promise.all([
             fetchDataWithRetry(() => getMyData()),
-            fetchDataWithRetry(() => loadQuestConfig()),
+            fetchDataWithRetry(() => getAppSettings(appSettingsArray)),
         ]);
 
-        if(!hasUserData(responseData)) {
+        if(!hasUserData(participantResponse)) {
             throw new Error('No user data found.');
         }
 
-        data = responseData.data;
+        if (!appSettingsResponse) {
+            throw new Error('Error fetching app settings.');
+        }
+
+        participantData = participantResponse.data;
+        appSettingsData = appSettingsResponse;
```

Comment:
Inconsistent style here. 
In future we may need to update `getMyData` function making it return participant data ( or null), so that there's no need to unwrap.